### ISSUE TITLE
Allow 'Extrapolate Speed' for negative values

### ISSUE THREAD

**Issue by Flavelius** [KybernetikGames/animancer#102]
# Use Case

I use _negative_ thresholds for backwards movement in a linear mixer and want larger values to influence playback speed.
For positive values this works as expected through 'Extrapolate Speed'.
for negative values it is clamped to the threshold value.

# Solution

Allowing negative values larger than the threshold to increase playback speed here like it's possible for possitive values would help.

**Comment by KybernetikGames** [comment]
Should be a simple enough modification to `LinearMixerState.ForceRecalculateWeights`.

```
var parameter = Parameter;
var previousThreshold = GetThreshold(index);

// Replace everything in here.

previousState.Weight = 1;
```

With this:

```
if (parameter <= previousThreshold)
{
    DisableRemainingStates(index);

    if (previousThreshold >= 0)
    {
        previousState.Weight = 1;
        goto ResetExtrapolatedSpeed;
    }
    // Here's the change. If it's negative, we just let it extrapolate normally.
}
else
{
    var childCount = ChildCount;
    while (++index < childCount)
    {
        var nextState = GetNextState(ref index);
        if (nextState == null)
            break;

        var nextThreshold = GetThreshold(index);

        if (parameter > previousThreshold && parameter <= nextThreshold)
        {
            var t = (parameter - previousThreshold) / (nextThreshold - previousThreshold);
            previousState.Weight = 1 - t;
            nextState.Weight = t;
            DisableRemainingStates(index);
            goto ResetExtrapolatedSpeed;
        }
        else
        {
            previousState.Weight = 0;
        }

        previousState = nextState;
        previousThreshold = nextThreshold;
    }
}
```

Let me know if that works for you.

**Comment by Flavelius** [comment]
Yes, that works. 
The goto's are a good candidate for local function + return btw

**Comment by KybernetikGames** [comment]
Maybe, but that would take a few more lines for a method and its comment and would disguise the fact that either way the method ends can lead to `_Playable.SetSpeed`.

**Comment by KybernetikGames** [comment]
This fix is included in Animancer v6.1 which is now available on [itch.io](https://kybernetik.itch.io/animancer-pro) and should be approved on the [Asset Store](https://assetstore.unity.com/packages/tools/animation/animancer-pro-116514?aid=1100l8ah5&utm_source=aff) in a few days.
